extensions/mssql/src/main.ts

```diff
@@ -87,6 +87,29 @@ export async function activate(context: vscode.ExtensionContext): Promise<IExten
 		}
 	});
 
+	vscode.commands.registerCommand(Constants.cmdObjectExplorerEnableGroupBySchemaCommand, async () => {
+		await vscode.workspace.getConfiguration().update(Constants.cmdObjectExplorerGroupBySchemaFlagName, true, true);
+	});
+
+	vscode.commands.registerCommand(Constants.cmdObjectExplorerDisableGroupBySchemaCommand, async () => {
+		await vscode.workspace.getConfiguration().update(Constants.cmdObjectExplorerGroupBySchemaFlagName, false, true);
+	});
+
+	vscode.workspace.onDidChangeConfiguration(async e => {
+		if (e.affectsConfiguration(Constants.cmdObjectExplorerGroupBySchemaFlagName)) {
+			TelemetryReporter.sendTelemetryEvent('objectExplorerGroupBySchemaChanged', {
+				groupBySchema: vscode.workspace.getConfiguration().get(Constants.cmdObjectExplorerGroupBySchemaFlagName)
+			});
+			const activeConnections = await azdata.objectexplorer.getActiveConnectionNodes();
+			activeConnections.forEach(async node => {
+				const connectionProfile = (await azdata.connection.getConnections()).filter(c => c.connectionId === node.connectionId)[0];
```

Comment:
You shouldn't need to call getConnections each iteration of the loop - just do it once at the beginning. 

(it's a shame that extensions have to do this though...not sure why we don't have at least the provider ID in the node itself as well)